### PR TITLE
refactor: first draft of converting to hooks

### PR DESCRIPTION
**Pull request by Enubia**
this pr is a first draft of converting everything to hooks

so far everything works as intended, the only file i didn't touch right now is the helpers.js since i honestly have no clue what's going on in the withHelpers function

@bsides maybe you can give me some insights there

additionally i've added eslint and prettier configurations, let me know if i should remove them, just thought it might be a good idea since the project wasn't really formatted tbh

i also updated all dependencies since most of them weren't up to date

### CODE REVIEW

**Comment by bsides** [comment]
Just by the description I'm already loving it. Will review your PR later today. Thank you so much for you efforts!

**Comment by Enubia** [comment]
maybe you could check out my other PRs before that since those contain fixes and features which I'd need to merge in this branch as well

**Review by bsides — COMMENTED** [review]
Noting to myself to review this weekend

**Comment by bsides** [comment]
> maybe you could check out my other PRs before that since those contain fixes and features which I'd need to merge in this branch as well

will do!

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (Enubia) [reply .eslintrc.js:1]: Well the IDE, doesn't matter if VSCode or WebStorm, doesn't format anything without a config. Yes there's a default prettier setting in VSCode but for example I don't have that enabled and it differs between developers what they have setup. Sure it is a lot of configuration stuff but CRA doesn't format without config and neither does husky/lint-staged.

I can remove it as I said ealier, just thought that some sort of styling guidline should be done.

**Pull request closed without merging**